src/imageLoader/wadors/getPixelData.js

```diff
@@ -29,9 +29,144 @@ function uint8ArrayToString(data, offset, length) {
   return str;
 }
 
-function getPixelData(uri, imageId, mediaType = 'application/octet-stream') {
+/**
+ * Asserts that a given media type is valid.
+ *
+ * @params {String} mediaType media type
+ */
+
+function assertMediaTypeIsValid(mediaType) {
+  if (!mediaType) {
+    throw new Error(`Not a valid media type: ${mediaType}`);
+  }
+  const sepIndex = mediaType.indexOf('/');
+
+  if (sepIndex === -1) {
+    throw new Error(`Not a valid media type: ${mediaType}`);
+  }
+  const mediaTypeType = mediaType.slice(0, sepIndex);
+
+  const types = ['application', 'image', 'text', 'video'];
+
+  if (!types.includes(mediaTypeType)) {
+    throw new Error(`Not a valid media type: ${mediaType}`);
+  }
+  if (mediaType.slice(sepIndex + 1).includes('/')) {
+    throw new Error(`Not a valid media type: ${mediaType}`);
+  }
+}
+
+/**
+ * Parses media type and extracts its type and subtype.
+ *
+ * @param mediaType e.g. image/jpeg
+ * @private
+ */
+function parseMediaType(mediaType) {
+  assertMediaTypeIsValid(mediaType);
+
+  return mediaType.split('/');
+}
+
+/**
+ * Builds an accept header field value for HTTP GET multipart request
+ messages.
+ *
+ * @param {Array} mediaTypes Acceptable media types
+ * @param {Object} supportedMediaTypes Supported media types
+ */
+
+function buildMultipartAcceptHeaderFieldValue(mediaTypes, supportedMediaTypes) {
+  if (!Array.isArray(mediaTypes)) {
+    throw new Error('Acceptable media types must be provided as an Array');
+  }
+
+  if (typeof supportedMediaTypes !== 'object') {
+    throw new Error(
+      'Supported media types must be provided as an Array or an Object'
+    );
+  }
+
+  const fieldValueParts = [];
+
+  mediaTypes.forEach((item) => {
+    const { transferSyntaxUID, mediaType } = item;
+
+    assertMediaTypeIsValid(mediaType);
+
+    let fieldValue = `multipart/related; type="${mediaType}"`;
+
+    // SupportedMediaTypes is a lookup table that maps Transfer Syntax UID
+    // to one or more Media Types
+    if (!Object.values(supportedMediaTypes).flat(1).includes(mediaType)) {
```

Comment:
this is not optimized, why we are doing Object.values again and again in the loop?

Reply:
you are right, fixed in https://github.com/cornerstonejs/cornerstoneWADOImageLoader/pull/477/commits/55628dd0ed4ef2b8d77cd6cb3394f9ba264ad5e6